Add: $removeCooldown, $wordCount and $editEmbed

SaiaAmako5 opened the pull request:
## Type
- [ ] Bug Fix
- [x] Functions: $removeCooldown, $editEmbed and $wordCount
- [ ] Callbacks: \<Callback Name>
- [ ] Handlers: \<Handler Type>
- [ ] Others: \______

Dependencies (Third Party Modules) needed: None. Not needed. 

Want a credit? Discord tag or other social media link: saiaamako | [saiaamako5](https://github.com/SaiaAmako5) 

Referenced Issue: #NaN (Answer for an issue, if any)

## Description
Adding three functions: $removeCooldown, $editEmbed and $wordCount that i made.

SaiaAmako5 committed:
Add: $editEmbed

Edits a certain embed option. 
Usage: $editEmbed[channelID;messageID;index;option;content]
Most option are the same as those of $getEmbed. 
timestamp option requires ms.

SaiaAmako5 committed:
Add: $wordCount

Counts the number of words in a given text. 
Usage: $wordCount[text;find?]

SaiaAmako5 committed:
Update wordCount.js

Now properly finds the count of a given word.

Faf4a commented on src/functions/guild/editEmbed.js:
```diff
@@ -0,0 +1,58 @@
+module.exports = async d => {
```
Use prettier and follow the project's styling.

SaiaAmako5 replied:
Used prettier. Please check it.

Faf4a commented on src/functions/guild/editEmbed.js:
```diff
@@ -0,0 +1,58 @@
+if (!isNumber(index)) return d.aoiError.fnError(d, 'custom', {}, 'Invalid Index');
+if (!content) return d.aoiError.fnError(d, 'custom', {}, 'No Content Provided');
+var embed = new d.embed(message.embeds[index]);
+if(option.startsWith("field[")) {
+const fi = option.charAt(6) - 1;
+option = option.slice(0, 6) + fi + option.slice(7); 
+option = option.slice(0, 5) + "s" + option.slice(5); 
+} else if(option.startsWith("field")) {
+const fi = option.charAt(5) - 1;
+option = option.slice(0, 5) + fi + option.slice(6); 
+option = option.slice(0, 5) + "[" + option.slice(5);
+option = option.slice(0, 7) + "]" + option.slice(7);
+option = option.slice(0, 5) + "s" + option.slice(5); 
+} else if(option == "timestamp") {
```
don't do this, this is actually unreadable, check parsers src for reference

SaiaAmako5 replied:
Ah i am sorry, I don't know why I did this, when I could have just used .substring(). Well I committed a new update for it. Please look into it.

Faf4a commented on src/functions/util/wordCount.js:
```diff
@@ -0,0 +1,13 @@
+
```
follow styling here as well

Faf4a commented on src/functions/guild/editEmbed.js:
```diff
@@ -0,0 +1,58 @@
+if (!channel) return d.aoiError.fnError(d, 'channel', {}, {inside: data.inside});
+const message = await d.util.getMessage(channel, messageID);
+if (!message) return d.aoiError.fnError(d, 'message', {}, {inside: data.inside});
+function isNumber(value) {
+  return typeof value === "number" && !Number.isNaN(value);
+}
```
just use isNaN(Number), aoijs uses the concept of strings therefore numbers dont exist unless converted into using Number() or similar iirc

SaiaAmako5 replied:
Okay

Faf4a reviewed (changes requested):
Thanks for the pull request and the useful additions, however think about applying the suggestions!

Commit:
Merge branch 'v6' into v6

SaiaAmako5 committed:
Delete src/functions/guild directory

According to new structure of function directory

SaiaAmako5 committed:
Delete src/functions/util directory

Deleted this directory as all the functions no longer are arranged in sub-directory.

SaiaAmako5 committed:
Update editEmbed.js

Removed the confusing field checker. Used a very basic one. Used prettier to make the code looks smooth.

SaiaAmako5 committed:
Update wordCount.js

Removed extra empty line.

SaiaAmako5 committed:
Update editEmbed.js

Fixed saying "invalid index".

Faf4a commented on src/functions/editEmbed.js:
```diff
@@ -0,0 +1,59 @@
+    }
+  }
+  try {
+    if (!eval(`embed.data.${option}`))
```
Please prevent using eval() directly, unless absolutely needed.

Faf4a commented on src/functions/editEmbed.js:
```diff
@@ -0,0 +1,59 @@
+      return d.aoiError.fnError(d, "custom", {}, "Invalid Option Provided");
+    eval(`embed.data.${option} = content.addBrackets()`);
+  } catch {
+    d.aoiError.fnError(d, "custom", {}, "Option/Index Not Found");
+    return;
```
Merge those into the same line

Faf4a commented on src/functions/editEmbed.js:
```diff
@@ -0,0 +1,59 @@
+      const ts = new Date(Number(content));
+      content = ts.toISOString();
+    } catch {
+      d.aoiError.fnError(d, "custom", {}, "Invalid MS Provided");
+      return;
```
Merge those in the same line

Faf4a commented on src/functions/editEmbed.js:
```diff
@@ -0,0 +1,59 @@
+    channelID = d.channel?.id,
+    messageID = d.message?.id,
+    index = 1,
+    option = "description",
```
Remove default option, probably not the best to have this especially when some embeds dont have a description

Faf4a commented on src/functions/editEmbed.js:
```diff
@@ -0,0 +1,59 @@
+  const message = await d.util.getMessage(channel, messageID);
+  if (!message)
+    return d.aoiError.fnError(d, "message", {}, { inside: data.inside });
+  if (isNaN(index))
+    return d.aoiError.fnError(d, "custom", {}, "Invalid Index");
```
this doesnt actually check for negative numbers, add a check for that

Faf4a commented on src/functions/editEmbed.js:
```diff
@@ -0,0 +1,59 @@
+    return d.aoiError.fnError(d, "custom", {}, "Invalid Index");
+  if (!content)
+    return d.aoiError.fnError(d, "custom", {}, "No Content Provided");
+  var embed = new d.embed(message.embeds[index]);
```
dont use var, use let

Faf4a commented on src/functions/editEmbed.js:
```diff
@@ -0,0 +1,59 @@
+  try {
+    if (!eval(`embed.data.${option}`))
+      return d.aoiError.fnError(d, "custom", {}, "Invalid Option Provided");
+    eval(`embed.data.${option} = content.addBrackets()`);
```
try to not use eval

Faf4a commented on src/functions/editEmbed.js:
Use the prettier config inside the root directory, prettier should detect it when using `ALT + SHIFT + F` or `prettier --w`!

Faf4a reviewed (changes requested):
*Few* changes before this can get merged! Let me know if you're still interested on working on these functions, sorry for the long wait!

SaiaAmako5 committed:
Merge branch 'aoijs:v6' into v6

SaiaAmako5 committed:
Update editEmbed.js

Changes done as asked.

SaiaAmako5 committed:
Update editEmbed.js

Ran prettier and fixed a typo.

Leref committed:
Merge branch 'v6' into v6

Faf4a reviewed (approved):
LGTM

SaiaAmako5 committed:
Update editEmbed.js

Fixed the editEmbed finally.

SaiaAmako5 committed:
Merge branch 'aoijs:v6' into v6

SaiaAmako5 committed:
Create removeCooldown.js

Removes a cooldown. 

Usage:
$removeCooldown[command;type;...id?]

 Parameters:
command | required | The name of the command from which you want to remove the cooldown from.
type | required | The type of the cooldown. (Available types: globalUser, user, guild, and channel)
...id?* | optional (auto selects the author) | The id of the User/guild/channel from the cooldown.

* Although the '...id?' is a rest parameter, but it is for the second ID for those type which supports it like 'user' so we can use it like this: `$removeCooldown[...;user;$authorID;$guildID]`

SaiaAmako5 committed:
Merge pull request #1 from SaiaAmako5/feat.-$removeCooldown

Create removeCooldown.js

Commit:
Merge branch 'v6' into v6

Faf4a commented on src/functions/removeCooldown.js:
```diff
@@ -0,0 +1,21 @@
+        channel: id[0] || d.message.channel.id
+    };
+    if (!types[type]) return d.aoiError.fnError(d, "custom", { inside: data.inside }, "Invalid Type In");
+    if (await d.client.db.get(type == "guild" ? d.client.db.tables[0] : "__aoijs_vars__", "cooldown", command + "_" + types[type])) {
```
Why?

Cooldowns are always stored into the internal `__aoijs_vars__`.

SaiaAmako5 replied:
You can check the `guildCooldown.js`, it seems to store the CD in the first db table instead of `__aoijs_vars__`.

Faf4a replied:
That's indeed unintended behaviour, which will be fixed in the next version. Therefore please change this to the `__aoijs_vars__` table.

SaiaAmako5 replied:
I see. I have committed a change. Please check it if it's okay.

Leref reviewed (commented):
I mean it's good beside @Faf4a stated of variable format.

SaiaAmako5 committed:
Merge branch 'aoijs:v6' into v6

SaiaAmako5 committed:
Update removeCooldown.js

To fix the variable. Changed to `__aoijs_vars__` for all `types`.

Leref commented:
Will close since no updates. Re-open if wish to maintain this PR.
